src/test/java/com/kodilla/ecommercee/domain/GroupEntityTests.java

```diff
@@ -0,0 +1,140 @@
+package com.kodilla.ecommercee.domain;
+
+import com.kodilla.ecommercee.repository.GroupRepository;
+import com.kodilla.ecommercee.repository.ProductRepository;
+import static org.junit.jupiter.api.Assertions.*;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import javax.transaction.Transactional;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@SpringBootTest
+@Transactional
+public class GroupEntityTests {
+
+    @Autowired
+    private GroupRepository groupRepository;
+
+    @Autowired
+    private ProductRepository productRepository;
+
+    @Test
+    public void testSaveGroup() {
+
+        //Given
+        Group group = new Group();
+
+        Product product = new Product();
+        List<Product> productList = new ArrayList<>();
+        productList.add(product);
+        productRepository.save(product);
+        group.setProducts(productList);
+        group.setName("Test Group");
```

Comment:
This code is repeted in every test, to make life easier You can select those 6 lines and right click - refactor - extract method or write it by yourself as a metod, then use method - one line of code, instad of copying or writing it in ecery test.
It's just a "let You know it's possible" kind of suggestion :)

Reply:
Thanks, next time I'll do like this :)